geometry-builder.cpp

```diff
@@ -363,7 +361,7 @@ geometry_builder::maybe_wkts_t geometry_builder::build_polygons(const osmNode *
         WKTWriter writer;
 
         // Procces ways into lines or simple polygon list
-        polygondata* polys = new polygondata[merged->size()];
+        std::vector<polygondata> polys(merged->size());
```

Comment:
If you do that (and I'm all in favour), it would be better to go the whole nine yards and write a constructor for polygondata from gf and merged[i] and then use push_back(). totalpolys can then be replaced by a simple polys.size(). Also means that you can reduce the code duplication with the same stuff below.